python/lsst/ts/envsensors/sensor/temperature_sensor.py

```diff
@@ -0,0 +1,100 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["TemperatureSensor"]
+
+import asyncio
+import logging
+import math
+from typing import List, Optional, Union
+
+from ..constants import DISCONNECTED_VALUE
+from ..response_code import ResponseCode
+from .base_sensor import BaseSensor, DELIMITER, TERMINATOR
+
+
+class TemperatureSensor(BaseSensor):
+    """Temperature Sensor.
+
+    Parameters
+    ----------
+    name: `str`
+        The name of the sensor.
+    channels: `int`
+        The number of temperature channels.
+    log: `logger`' optional
+        The logger for which to create a child logger, or None in which case a
+        new logger gets requested.
+    """
+
+    def __init__(
+        self,
+        channels: int,
+        log: logging.Logger,
+    ) -> None:
+        super().__init__(channels=channels, log=log)
+
+    async def open(self) -> None:
+        """Open a connection to the Sensor and set parameters."""
+        pass
+
+    async def extract_telemetry(self, line: str) -> List[float]:
+        """Extract the temperature telemetry from a line of Sensor data.
+
+        Parameters
+        ----------
+        line: `str`
+            A line of comma separated telemetry, each of the format
+            CXX=XXXX.XXX
+
+        Returns
+        -------
+        output: `list`
+            A list containing the temperature telemetry as measured by the
+            specific type of sensor. The length of the output list is the same
+            as the number of channels.
+            If a channel is disconnected (its value will be DISCONNECTED_VALUE)
+            or if a channel is missing then the value gets replaced by math.nan
+        """
+        self._log.debug("extract_telemetry")
+        stripped_line: str = line.strip(TERMINATOR)
+        line_items = stripped_line.split(DELIMITER)
+        output = []
+        for line_item in line_items:
+            temperature_items = line_item.split("=")
+            if len(temperature_items) == 1:
+                output.append(math.nan)
+            elif len(temperature_items) == 2:
+                if temperature_items[1] == DISCONNECTED_VALUE:
+                    output.append(math.nan)
+                else:
+                    output.append(float(temperature_items[1]))
+            else:
+                raise ValueError(
+                    f"At most one '=' symbol expected in temperature item {line_item}"
+                )
+        while len(output) < self.channels:
+            output.insert(0, math.nan)
```

Comment:
A code comment explaining what's going on here would be very helpful. Are missing values always from the beginning? I find that really surprising.

If the missing values can be from any channel then a safer way to handle this is to first make a dict of `channel: value` for the values you DID read (where channel could be a str or int). Then assemble the final list by iterating over all valid channel numbers. That also makes it easy to handle missing channels and even to handle unexpected/incorrect channels. E.g. something along the line of:
```
# assuming self.channel_list is something you create in the constructor
output = [output_dict.pop(channel, np.nan) for channel in self.channel_list]
if output_dict:
   handle unexpected channels
```

Reply:
The sensor starts outputting data as soon as it is switched on, not as soon as a connection is made. I added a comment in the code to clarify that.